tests/category-endpoints/category.test.ts

```diff
@@ -0,0 +1,605 @@
+          }, {
+            field: 'color',
+            validationCode: 'color.LENGTH_NOT_VALID'
+          })
+        })
+
+        it('verifies that the string is not empty', async () => {
+          await runTest({
+            ...baseResponse,
+            color: ''
+          }, {
+            field: 'color',
+            validationCode: 'color.LENGTH_NOT_VALID'
+          })
+        })
+      })
+
+      describe('Returns 400 when `extra` is invalid', () => {
+        const baseResponse = {
+          type: 'TO_USER',
+          pictureUrl: 'http://google.com',
+          description: 'Sample description',
+          name: 'Test name',
+          color: '#000000'
+        }
+
+        it('verifies that the field is a string', async () => {
+          await runTest({
+            ...baseResponse,
+            extra: 1
+          }, {
+            field: 'extra',
+            validationCode: 'extra.STRING'
+          })
+        })
+
+        it('verifies that the string doesn\'t exceed 255 chars', async () => {
+          let extra = 'abcde'
+          while (extra.length < 256) {
+            extra += extra
+          }
+
+          await runTest({
+            ...baseResponse,
+            extra: extra
+          }, {
+            field: 'extra',
+            validationCode: 'extra.LENGTH_NOT_VALID'
+          })
+        })
+      })
+    })
+  })
+
+  describe('Category lookup by ID', () => {
+    const url = `${baseUrl}/categories`
+
+    it('Allows for a category to be gotten', async () => {
+      await db.seed.run()
+      const expected : CreateCategoryRequest = {
+        name: 'Testing Category Get',
+        type: 'TO_USER',
+        pictureUrl: 'http://google.com',
+        description: 'Sample description',
+        extra: 'Extra information',
+        color: '#000000'
+      }
+
+      const resCreation = await request(app)
+        .post(url)
+        .send(expected)
+
+      expect(resCreation.status).toBe(201)
+
+      const id = resCreation.body[0].id
+      expect(id).toBeDefined()
+
+      const lookupUrl = url + `/${id}`
+      const res = await request(app)
+        .get(lookupUrl)
+      expect(res.status).toBe(200)
+
+      const category = {
+        id: res.body.id,
+        name: res.body.name,
+        type: res.body.type,
+        pictureUrl: res.body.pictureUrl,
+        description: res.body.description,
+        extra: res.body.extra,
+        color: res.body.color,
+      }
+
+      expect(category.id).toBe(id)
+      expect(category.name).toBe(expected.name)
+      expect(category.type).toBe(expected.type)
+      expect(category.pictureUrl).toBe(expected.pictureUrl)
+      expect(category.description).toBe(expected.description)
+      expect(category.extra).toBe(expected.extra)
+      expect(category.color).toBe(expected.color)
+    })
+
+    it('Raises a not found error on invalid id', async() => {
+      const id = -1
+      const lookupUrl = url + `/${id}`
```

Comment:
This could be rewritten as this:
```suggestion
      const lookupUrl = `${url}/${id}`
```